### PR TITLE
Documentation for Data Prepper's kafka buffer

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

This adds a new page to the Data Prepper documentation for the new `kafka` buffer available in Data Prepper 2.6.0.

### Issues Resolved

N/A

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Naarcha-AWS** [comment]
@dlvenable: Do you have a usage example for the `kafka` buffer?

**Comment by dlvenable** [comment]
@Naarcha-AWS, I added an example pipeline configuration. Good suggestion.

File: _data-prepper/pipelines/configuration/buffers/kafka.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+Use the following options when setting SSL encryption.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`type` | No | String | The encryption type. Use `none` to disable encryption. Default is `ssl`.
+`insecure` | No | Boolean | A Boolean flag used to turn off SSL certificate verification. If set to `true`, certificate authority (CA) certificate verification is turned off and insecure HTTP requests are sent. Default is `false`.
+
+
+#### AWS
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/buffers/kafka.md:127]: Should this be "aws"?

File: _data-prepper/pipelines/configuration/buffers/kafka.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+Use the following options when setting up authentication for `aws` services.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`region` | No | String | The AWS Region to use for credentials. Defaults to [standard SDK behavior to determine the Region](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html).
+`sts_role_arn` | No | String | The AWS Security Token Service (AWS STS) role to assume for requests to Amazon Simple Queue Service (Amazon SQS) and Amazon Simple Storage Service (Amazon S3). Default is `null`, which will use the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html).
+`msk` | No | JSON object | The [MSK](#msk) configuration settings.
+
+#### MSK
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/buffers/kafka.md:137]: Should this be "msk"?

File: _data-prepper/pipelines/configuration/buffers/kafka.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+Use the following options inside the `msk` object.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`arn` | Yes | String | The [MSK ARN](https://docs.aws.amazon.com/msk/1.0/apireference/configurations-arn.html) to use.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/buffers/kafka.md:143]:
```suggestion
`arn` | Yes | String | The [Amazon MSK ARN](https://docs.aws.amazon.com/msk/1.0/apireference/configurations-arn.html) to use.
```

File: _data-prepper/pipelines/configuration/buffers/kafka.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+`broker_connection_type` No | String | The type of connector to use with the MSK broker, either `public`, `single_vpc`, or `multi_vpc`. Default is `single_vpc`.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/buffers/kafka.md:144]:
```suggestion
`broker_connection_type` No | String | The type of connector to use with the Amazon MSK broker, either `public`, `single_vpc`, or `multi_vpc`. Default is `single_vpc`.
```

**Review by natebower — CHANGES_REQUESTED** [review]
@Naarcha-AWS Please see my comments and changes and let me know if you have any questions. I'd like to read line 53 again before approving, so please tag me when complete. Thanks!

File: _data-prepper/pipelines/configuration/buffers/kafka.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+`bootstrap_servers` | Yes | String list | The host and port for the initial connection to the Kafka cluster. You can configure multiple Kafka brokers by using the IP address or the port number for each broker. When using [Amazon MSK](https://aws.amazon.com/msk/) as your Kafka cluster, the bootstrap server information is obtained from MSK using the MSK Amazon Resource Name (ARN) provided in the configuration.
+`topics` | Yes | List | A list of [topics](#topic) to use. You must supply one topic per buffer.
+`authentication` | No | [Authentication](#authentication) | Sets the authentication options for both the pipeline and Kafka. For more information, see [Authentication](#authentication).
+`encryption` | No | [Encryption](#encryption) | The encryption configuration for encryption-in-transit. For more information, see [Encryption](#encryption).
+`aws` | No | [AWS](#aws) | The AWS configuration. For more information, see [aws](#aws).
+
+
+### topic
+
+Each `topic` configures a single Kafka topic and how to use that Kafka topic.
```
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/configuration/buffers/kafka.md:53]:
```suggestion
The `topic` option configures a single Kafka topic and tells the `kafka` buffer use that topic.
```

Comment (Naarcha-AWS) [reply _data-prepper/pipelines/configuration/buffers/kafka.md:53]: Does this work @natebower?

File: _data-prepper/pipelines/configuration/buffers/kafka.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+`plaintext` | JSON object | The [PLAINTEXT](#sasl-plaintext) authentication configuration.
+`aws_msk_iam` | String | The Amazon MSK AWS Identity and Access Management (IAM) configuration. If set to `role`, the `sts_role_arm` set in the `aws` configuration is used. Default is `default`.
+
+#### SASL PLAINTEXT
+
+The following options are required when using the [SASL PLAINTEXT](https://kafka.apache.org/10/javadoc/org/apache/kafka/common/security/auth/SecurityProtocol.html) protocol.
+
+Option | Type | Description
+:--- | :--- | :---
+`username` | String | The username for the PLAINTEXT auth.
```
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/configuration/buffers/kafka.md:114]:
```suggestion
`username` | String | The username for the PLAINTEXT authentication.
```

File: _data-prepper/pipelines/configuration/buffers/kafka.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+`password` | String | The password for the PLAINTEXT auth.
```
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/configuration/buffers/kafka.md:115]:
```suggestion
`password` | String | The password for the PLAINTEXT authentication.
```

File: _data-prepper/pipelines/configuration/buffers/kafka.md (hunk 1 above)
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/configuration/buffers/kafka.md:127]:
```suggestion
#### aws
```

File: _data-prepper/pipelines/configuration/buffers/kafka.md (hunk 2 above)
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/configuration/buffers/kafka.md:137]:
```suggestion
#### msk
```

File: _data-prepper/pipelines/configuration/buffers/kafka.md (hunk 5 above)
Comment (natebower) [reply _data-prepper/pipelines/configuration/buffers/kafka.md:53]: Yes, but "to" before "use".

**Review by natebower — APPROVED** [review]
LGTM

**Pull request merged**